### ISSUE TITLE
Ignores comment after generic JsxOpeningElement

### ISSUE THREAD

**Issue by duailibe** [typescript-eslint/typescript-eslint#583]
**What code were you trying to parse?**

```ts
const comp = (
  <Component<number> /* Foo */></Component>
);
```

**What did you expect to happen?**

A comment node "Foo" should be included somewhere

**What actually happened?**

Comment "Foo" is ignored and this affects Prettier making we drop comments (see: https://github.com/prettier/prettier/issues/6012).

I'm not familiar with TypeScript parser API, but I tested a few things and if I comment the following `case` condition, it works:

https://github.com/typescript-eslint/typescript-eslint/blob/50a493e4c98f06b768aa9234c29e9da2630b2fc4/packages/typescript-estree/src/convert-comments.ts#L118-L130

Maybe that helps to identify the problem.

I'm willing to contribute with a fix if you have your hands full, but I need to learn what that code does. Let me know if you need help.

**Versions**

Tested in master.

Pinging @JamesHenry

**Comment by bradzacher** [comment]
Feel free to submit a PR if you can.
We have good test coverage so it makes it easy to make changes without full knowledge of the codebase.

**Comment by kaicataldo** [comment]
Is this actually valid JSX? I thought JS comments had to be inside curly braces in JSX.

**Comment by bradzacher** [comment]
Having a look at the react docs, they don't say anything specifically about comments within a JSX tag.
Reading the spec (https://facebook.github.io/jsx/) it says (very vaguely):
> JSX uses the same punctuators and braces as ECMAScript. WhiteSpace, LineTerminators and Comments are generally allowed between any punctuators.

Looking in astexplorer, all of the babel parsers seem to handle it.

We should probably look for parity with babel in this regard, if only because I assume the react team contributes to that portion of the parser and so is a defacto standard.

**Comment by kaicataldo** [comment]
In the examples given [here](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-comment-textnodes.md), it does look like comments can exist within a tag. The curly braces are only needed to ensure it doesn’t get rendered as a text node. TIL!
